packages/vite/src/node/__tests__/plugins/css.spec.ts

```diff
@@ -258,3 +261,142 @@ describe('convertTargets', () => {
+        code: '',
+        imports: [],
+        viteMetadata: {
+          importedAssets: new Set<string>(),
+          importedCss: new Set<string>(),
+        },
+      } as any as OutputChunk,
+    }
+
+    removePureCssChunks(bundle, ['pure_css_chunk.js'], 'es')
+
+    const chunk = bundle['main.js'] as OutputChunk
+    expect(chunk.code).toMatchSnapshot()
+    // import is removed
+    expect(chunk.imports).toEqual(['some-module', 'other-module'])
+  })
+
+  test('require of removed chunk is dropped', () => {
+    const bundle: OutputBundle = {
+      'main.js': {
+        code: 'require("some-module");\nrequire("pure_css_chunk.js");\nrequire("other-module");\n',
+        imports: ['pure_css_chunk.js', 'some-module', 'other-module'],
+        type: 'chunk',
+        viteMetadata: {
+          importedAssets: new Set<string>(),
+          importedCss: new Set<string>(),
+        },
+      } as any as OutputChunk,
+      'pure_css_chunk.js': {
+        type: 'chunk',
+        code: '',
+        imports: [],
+        viteMetadata: {
+          importedAssets: new Set<string>(),
+          importedCss: new Set<string>(),
+        },
+      } as any as OutputChunk,
+    }
+
+    removePureCssChunks(bundle, ['pure_css_chunk.js'], 'cjs')
+
+    const chunk = bundle['main.js'] as OutputChunk
+    expect(chunk.code).toMatchSnapshot()
+    // import is removed
+    expect(chunk.imports).toEqual(['some-module', 'other-module'])
+  })
+
+  test('imported assets of css chunk are transfered', () => {
+    const bundle: OutputBundle = {
+      'main.js': {
+        code: 'import "some-module";\nimport "pure_css_chunk.js";\nimport "other-module";\n',
+        imports: ['pure_css_chunk.js', 'some-module', 'other-module'],
+        type: 'chunk',
+        viteMetadata: {
+          importedAssets: new Set<string>(),
+          importedCss: new Set<string>(),
+        },
+      } as any as OutputChunk,
+      'pure_css_chunk.js': {
+        type: 'chunk',
+        code: '',
+        imports: [],
+        viteMetadata: {
+          importedAssets: new Set<string>(['some-asset.svg']),
+          importedCss: new Set<string>(['some-style.css']),
+        },
+      } as any as OutputChunk,
+    }
+
+    removePureCssChunks(bundle, ['pure_css_chunk.js'], 'es')
+
+    const chunk = bundle['main.js'] as OutputChunk
+    expect(chunk.code).toMatchSnapshot()
+    // import is removed
+    expect(chunk.imports).toEqual(['some-module', 'other-module'])
+    // metadata is transfered
+    expect(chunk.viteMetadata?.importedAssets.has('some-asset.svg')).toBe(true)
+    expect(chunk.viteMetadata?.importedCss.has('some-style.css')).toBe(true)
+  })
+})
+
+describe('getEmptyChunkReplacer', () => {
+  test('replaces import call', () => {
+    const code =
+      'import "some-module";\nimport "pure_css_chunk.js";\nimport "other-module";'
+
+    const replacer = getEmptyChunkReplacer(['pure_css_chunk.js'], 'es')
+    expect(replacer(code)).toMatchSnapshot()
```

Comment:
Minor nit, we usually have small snippet transforms using `toMatchInlineSnapshot`. It's easier to see what's the before and after closely. I know there's an outlier in `parse.test.ts.snap`, but I think that can be inline also in the future.